File: day4/compiled-codes/04-f2py/ex04/sum.f90

function sum_of_ints(a,n) result(s)
    integer, intent(in) :: a(*), n
    integer :: s, i

    s = 0
    do i=1,n
        s = s + a(i)
    end do
    print*, "sum of ", n, " elements is ", s
end function sum_of_ints


function sum_of_doubles(a,n) result(s)
    double precision, intent(in) :: a(*)
    integer, intent(in) :: n
    double precision :: s
    integer :: i

    s = 0
    do i=1,n
        s = s + a(i)
    end do
    print*, "sum of ", n, " elements is ", s
end function sum_of_doubles

File: day4/compiled-codes/04-f2py/ex04/sum.py
#!/usr/bin/env python

from math import *
from sum import *

# fill data structure
num = 10
ilist = [i-1 for i in range(1,num)]

print ("Calling fortran with ints")
isum = sum_of_ints(ilist,num)
print ("Integer sum is: ", isum)

print ("Calling fortran double sum with integers")
dsum = sum_of_doubles(ilist,num)
print ("Double sum of ints is: ", dsum)

# fill data structure
dlist = [sqrt(float(i)) for i in range(1,num)]

print ("Calling fortran with doubles")
dsum = sum_of_doubles(dlist,num)
print ("Double sum is: ", dsum)
